akka-stream-tests/src/test/scala/akka/stream/io/InputStreamSinkSpec.scala

```diff
@@ -275,6 +275,17 @@ class InputStreamSinkSpec extends StreamSpec(UnboundedMailboxConfig) {
       }
       thrown.getCause should ===(error)
     }
+
+    "a read of length 0 should not request bytes from upstream" in assertAllStagesStopped {
+      val (probe, inputStream) = TestSource.probe[ByteString].toMat(StreamConverters.asInputStream())(Keep.both).run()
+      val f = Future(inputStream.read(new Array[Byte](byteString.size), 0, 0))
+
+      probe.executeAfterSubscription(() => probe.expectNoMessage())
+      Await.result(f, remainingOrDefault) should ===(0)
+
+      probe.sendComplete()
+      inputStream.close()
```

Comment:
Already looking quite good! Indeed perhaps the test could be a bit more direct, like:

```suggestion
      probe.ensureSubscription()
      probe.expectRequest()
      
      inputStream.read(new Array[Byte](byteString.size), 0, 0) should ===(0)

      probe.expectNoMessage()

      inputStream.close()
      
      probe.expectCancellation()
```

Reply:
Thanks I simplified the the test